### PR TITLE
[enhancement](Nereids): refactor eliminating inner join by foreign key

### PR DESCRIPTION
**Pull request by keanji-x**
## Proposed changes

refactor eliminating inner join by foreign key

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/EliminateJoinByFK.java [hunk 1]
```diff
@@ -143,47 +127,56 @@ private Plan applyNullCompensationFilter(Plan child, Set<Slot> childSlots) {
...
+                builder.put(primarySlots.iterator().next(), foreignSlot);
+            }
+            return builder.build();
         }
 
         // Right now we only support eliminate inner join, which should meet the following condition:
         // 1. only contain null-reject equal condition, and which all meet fk-pk constraint
         // 2. only output foreign table output or can be converted to foreign table output
-        // 3. foreign key is unique
         // 4. if foreign key is null, add a isNotNull predicate for null-reject join condition
```
Comment (xzj7019) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/EliminateJoinByFK.java:164]: should be 3.

**Review by xzj7019 — APPROVED** [review]

**Review by jackwener — DISMISSED** [review]

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

**Review by jackwener — APPROVED** [review]

**Pull request merged**